### PR TITLE
Implement FEEL support for text views

### LINKED ISSUE #436: Allow specifying text view content via FEEL expression
__Is your feature request related to a problem? Please describe__

With #385 we enable properties to be FEEL optional. This opens up several chances to make form fields more flexible via input data and expressions. 

One use case is the `text` component content, which can be static or dynamically filled via an expression.

image

image

__Describe the solution you'd like__

Enable FEEL support for `Text`.

* [ ] replace Text properties panel entry with a FEEL optional textarea
* [ ] show editor placeholder if a FEEL expression is specified
* [ ] show editor placeholder if nothing is specified
* [ ] rename "Text" to "Text view" to be consistent with image views

__Describe alternatives you've considered__

/

__Additional context__

Child of https://github.com/camunda/team-hto/issues/79
Related (but not necessarily the same) to https://github.com/bpmn-io/form-js/issues/247

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #436 

* moves icons to viewer package
* allows specifying text content via FEEL
* renames `Text` to `Text view`

Check out [the demo](https://demo-436-text-component-feel--camunda-form-playground.netlify.app/).

image

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
This is ready for review now. 👍

**Comment by pinussilvestrus** [comment]
[`product` is a built-in function](https://docs.camunda.io/docs/components/modeler/feel/builtin-functions/feel-built-in-functions-list/#product), so declaring the expression `=product` would return the function itself.

image

We don't allow shadowing (overriding) keywords (cf. https://github.com/bpmn-io/form-js/issues/401#issuecomment-1326221363) so I think we have to live with that as of now /cc @christian-konrad

**Comment by pinussilvestrus** [comment]
We could create an issue to add linting rules for this (using keywords in expressions). We have the same problem in BPMN I'd assume.

**Comment by pinussilvestrus** [comment]
@Skaiir @vsgoulart do you think we can have a review for this one this week?

**Comment by christian-konrad** [comment]
@pinussilvestrus 

> We could create an issue to add linting rules for this (using keywords in expressions). We have the same problem in BPMN I'd assume.

Have you created that issue for the core team (i.e. for @marstamm)? I think it can become a hard-to-debug problem for users. 
For me, this sounds like a follow-up issue to the https://github.com/camunda/product-hub/issues/280 epic

**Comment by pinussilvestrus** [comment]
I didn't create an issue yet. Not sure where actually; maybe @marstamm has an idea.

File: packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js [hunk 1]
```diff
@@ -201,16 +324,17 @@ function createText(options = {}) {
     value
   } = options;
 
-  return render(
+  return render(WithFormContext(
```
Comment (vsgoulart) [inline packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js:327]: Why is this necessary?

File: packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js [hunk 2]
```diff
@@ -123,6 +125,104 @@ Some _em_ **strong** [text](#text) \`code\`.
...
+
+    // then
+    const formField = container.querySelector('.fjs-form-field');
+
+    expect(formField).to.exist;
+    expect(formField.innerHTML).to.eql('<div class="markup"><div xmlns="http://www.w3.org/1999/xhtml"><h1>foo,###bar</h1></div></div>');
+  });
+
+
+  describe('disabled', function() {
```
Comment (vsgoulart) [inline packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js:176]: On the Operate team we created a rule to avoid nested describes so it's easier to read and it's helps with keeping the tests isolated.

If you and @Skaiir like the idea we could consider doing this here too

File: packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js (hunk 1 above)
Comment (pinussilvestrus) [reply packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js:327]: The FormContext is needed to evaluate FEEL expressions, e.g. it's loading the full form data state into a single component.

In the form field unit tests setup, we mock it to encapsulate the test logic. If we would not mock it, it would simply blow up.

File: packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js (hunk 2 above)
Comment (pinussilvestrus) [reply packages/form-js-viewer/test/spec/render/components/form-fields/Text.spec.js:176]: I'm not sure actually. I'm in favor of nested describe blocks, to structure them logically. I guess this just personal taste, no strong feelings though.

File: packages/form-js-viewer/assets/form-js.css [hunk 3]
```diff
@@ -99,6 +99,15 @@
 
 .fjs-container .fjs-form-field-placeholder {
   color: var(--color-text-light);
+	font-style: italic;
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js-viewer/assets/form-js.css:102]: Will adjust :+1:

[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js-viewer/assets/form-js.css]: Resolved.

**Review by vsgoulart — APPROVED** [review]
